chart/templates/server-deployment.yaml

```diff
@@ -130,14 +154,46 @@ spec:
         {{- if $root.Values.clamav.enabled  }}
         - name: clamav
           image: "{{ $root.Values.clamav.image.value }}:{{ $root.Values.clamav.image.tag }}"
-          imagePullPolicy: {{ $root.Values.image.pullPolicy }}
+          imagePullPolicy: {{ $root.Values.clamav.image.pullPolicy }}
           ports:
-{{ toYaml $root.Values.clamav.ports | indent 12 }}
-{{ toYaml $root.Values.clamav.probes | indent 10 }}
+            - containerPort: 3310
+              name: clamavport
+              protocol: TCP
+          readinessProbe:
+            tcpSocket:
+              port: clamavport
+            failureThreshold: {{ $root.Values.clamav.readinessProbe.failureThreshold }}
+            initialDelaySeconds: 30
+            periodSeconds: {{ $root.Values.clamav.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ $root.Values.clamav.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ $root.Values.clamav.readinessProbe.successThreshold }}
+          startupProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  clamscan --version || echo "ClamAV failed to start"
+                  echo -n "clamav.startup.success:1|c" | nc -u -w1 localhost 8125
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            failureThreshold: 10
+            timeoutSeconds: 30
+            successThreshold: 1
+          lifecycle:
+            postStart:
+              exec:
+                command:
+                  - /bin/sh
+                  - -c
+                  - freshclam --config-file /etc/clamav/freshclam.conf
           resources:
-{{ toYaml $root.Values.clamav.resources | indent 12 }}
+            requests:
+              cpu: "100m"
```

Comment:
I'd recommend that the CPU and memory be abstracted into the values.yaml

Reply:
I'm making the change - but normally we would want to abstract out values that we fully expect will need to have variation between environments. This value is in the grey area.